main.py

```diff
@@ -2,61 +2,77 @@
 from services import Services
 from application import App
 from PyQt5.QtWidgets import QApplication
-
+import logging
+from datetime import date, timedelta
+import screen_brightness_control as sbc
+import os
+import stat
 
 BASE_URL = 'https://api.olhar.media/'
-
+BASE_DIRECTRY = './assets/logs/'
 service = Services()
 
-logger = service.get_logger('main')
+current_city = service.get_current_city()
 
-def main():
-    # try:
-    #     logger.info('Fetching script...')
-    #     script = service.fetch_script(f'{BASE_URL}?getconfigupdate&equipid=1')
-    #     if script:
-    #         if service.is_valid_bash(script):
-    #             with open('script.sh', 'wb') as f:
-    #                 f.write(script.encode())
-    #             service.run_bash('./script.sh')
-    #             os.remove('./script.sh')
-    #             logger.info('Script executed successfully')
-    #         elif service.is_valid_python(script):
-    #             with open('script.py', 'wb') as f:
-    #                 f.write(script.encode())
-    #             service.run_python('./script.py')
-    #             os.remove('./script.py')
-    #             logger.info('Script executed successfully')
-    # except Exception as e:
-    #     logger.error(e)
 
-    try:   
-        logger.info('Fetching video data...')
-        return service.fetch_json(f'{BASE_URL}?getvideos&equipid=1')
+logging.basicConfig(filename=f'{BASE_DIRECTRY}{date.today()}.log', level=logging.DEBUG)
+logger = logging.getLogger('main')
```

Comment:
1. You definitely initialize `service` it in your `main()`.
2. You keep `logger = logging.getLogger('main')` here
3. You call `logging.basicConfig` from main
4. You do not have `current city` in the global scope. you ask `services` whenever you need it.

Reply:
But then I get an error that logger (nor service) is not defined in the function `run _maintenance_script`. Should this be also a method of serivce?